src/pipelines/floodscan_pipeline.py

```diff
@@ -393,6 +397,48 @@ def run_pipeline(self):
             self.combine_bands(sfed_da, mdfed_da, latest_date)
             return True
 
+        elif self.baseline_update:
```

Comment:
I think would be good to put this in a separate function, so that we have a better separation of concerns within this `run_pipeline` function (and so it's more testable). Could we create a separate `generate_baseline` function inside this class?